### DISCUSSION TITLE
Serial Commands Reference for Coordinator Firmware

### DISCUSSION THREAD

**Discussion by rmeissne** [Koenkk/zigbee2mqtt#13445] [Devices]
Hi,

Is there some reference for all available serial commands please?

Like this one: fe00210223 which will query the firmware version with a possible result of:

fe0e61020200020603ab653401020000000091

This is basically the information which will be translated into the following log message:

Coordinator firmware version: '{"meta":{"maintrel":3,"majorrel":2,"minorrel":6,"product":0,"revision":20211115,"transportrev":2},"type":"zStack12"}'

-> ab653401 is the hex representation of the revision number 20211115

I am basically looking for a reference of all commands that can be send, what is the format/values and what are possible answers from the coordinator firmware.

Regards,
Robert

**Comment by Koenkk** [comment]
The ZNP api is described in https://github.com/Koenkk/zigbee-herdsman/blob/master/docs/Z-Stack%20Monitor%20and%20Test%20API.pdf

**Reply by rmeissne** [reply]
Thanks a lot!

**Reply by rmeissne** [reply]
There is a newer document:

https://dev.ti.com/tirex/explore/node?node=ACfNy3SGNWAy9KyeAluYLQ__BSEc4rl__LATEST

However, this still does not seem to be complete. I have the following command/response in my capture:

CMD: fe04211c82000000bb
Response: fe17611c00150001030507090b0d0f00020406080a0c0d4c1d00002d (no idea what this is)

CMD length is 0x04, cmd0 = 0x21, cmd1 = 0x1c

I can see a lot of "fe04211c ..." commands in my capture.

Does anybody maybe know a more up to date document or where else to look? I mean, this must be documented somewhere.

Thanks,
Robert
